src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -52,37 +65,97 @@ class AssetsTable extends React.Component {
     });
   }
 
+  modalClose() {
+    this.setState({ modalOpen: false });
+  }
+
+  deleteAsset() {
+    this.props.deleteAsset(this.props.assetsParameters, this.state.assetToDelete.id);
+    this.setState({ modalOpen: false });
+  }
+
+  renderModal() {
+    return (
+      <Modal
+        open={this.state.modalOpen}
+        title={`Delete ${this.state.assetToDelete.display_name}`}
+        body={this.renderBody()}
+        closeText="Cancel"
+        onClose={this.modalClose}
+        buttons={[
+          <Button
+            display="Yes, delete."
+            buttonType="primary"
+            onClick={this.deleteAsset}
+          />,
+        ]}
+      />
+    );
+  }
+
+  renderBody() {
+    return (
+      <div>
+        <span className={classNames(FontAwesomeStyles.fa, FontAwesomeStyles['fa-exclamation-triangle'])} aria-hidden="true" />
+        Are you sure you wish to delete this item?
+        This cannot be reversed! Any content that links/refers to this item will no longer work
+        (e.g. images and/or links may break).
+      </div>
+    );
+  }
+
   render() {
     return (!this.props.assetsList.length) ? (
       <span>Loading....</span>
     ) : (
-      <Table
-        columns={[
-          {
-            label: 'Name',
-            key: 'display_name',
-          },
-          {
-            label: 'Type',
-            key: 'content_type',
-          },
-          {
-            label: 'Date Added',
-            key: 'date_added',
-          },
-          {
-            label: 'Delete Asset',
-            key: 'delete_asset',
-          },
-        ]}
-        data={this.state.displayAssetsList}
-      />
+      <div>
+        <Table
+          columns={[
+            {
+              label: 'Name',
+              key: 'display_name',
+            },
+            {
+              label: 'Type',
+              key: 'content_type',
+            },
+            {
+              label: 'Date Added',
+              key: 'date_added',
+            },
+            {
+              label: 'Delete Asset',
+              key: 'delete_asset',
+            },
+          ]}
+          data={this.state.displayAssetsList}
+        />
+        {this.renderModal()}
+      </div>
     );
   }
 }
 
 AssetsTable.propTypes = {
   assetsList: PropTypes.arrayOf(PropTypes.object).isRequired,
+  assetsParameters: PropTypes.objectOf(
+    PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool, PropTypes.object]),
+  ).isRequired,
+  deleteAsset: PropTypes.func.isRequired,
 };
 
-export default AssetsTable;
+const mapStateToProps = state => {
```

Comment:
I pulled out the mapStateToProps and mapDispatchToProps functions from the connect function call, since I thought it was a little more clear what the connect function was doing this way. When I was looking at WrappedAssetsFilters as an example, it took me some time to realize what the connect function was doing, so I think that giving the functions explicit "Google-able" names may be helpful for people unfamiliar with react-redux, instead of them being defined inside the connect call. Any thoughts?